d2/src/main/java/com/linkedin/d2/balancer/dualread/DualReadLoadBalancerMonitor.java

```diff
@@ -71,29 +71,37 @@ public void reportData(String propertyName, T property, String propertyVersion,
     Cache<String, CacheEntry<T>> cacheToAdd = fromNewLb ? _newLbPropertyCache : _oldLbPropertyCache;
     Cache<String, CacheEntry<T>> cacheToCompare = fromNewLb ? _oldLbPropertyCache : _newLbPropertyCache;
 
-    CacheEntry<T> entry = cacheToCompare.getIfPresent(propertyName);
+    CacheEntry<T> entryToCompare = cacheToCompare.getIfPresent(propertyName);
+    CacheEntry<T> newEntry = new CacheEntry<>(propertyVersion, getTimestamp(), property);
 
-    if (entry != null && Objects.equals(entry._version, propertyVersion))
+    if (entryToCompare != null && Objects.equals(entryToCompare._version, propertyVersion))
     {
-      CacheEntry<T> newEntry = new CacheEntry<>(propertyVersion, getTimestamp(), property);
-      if (!isEqual(entry, newEntry))
+      if (!isEqual(entryToCompare, newEntry))
       {
         _rateLimitedLogger.warn("Received mismatched properties from dual read. Old LB: {}, New LB: {}",
-            fromNewLb ? entry : newEntry, fromNewLb ? newEntry : entry);
+            fromNewLb ? entryToCompare : newEntry, fromNewLb ? newEntry : entryToCompare);
```

Comment:
Too bad this logic is repeated in multiple places... I can't think of a good way to simplify it though, so I guess it's fine to leave it as is unless you can think of something?